plan/logical_plans.go

```diff
@@ -77,6 +77,7 @@ type Selection struct {
 	// split into a list of AND conditions.
 	Conditions []expression.Expression
 
+	// aboveTableScan means if this selection's child is a table scan or index scan.
```

Comment:
So `aboveTableScan` is false if the selection's child is a join or having? Then `above` is not accurate, and `TableScan` doesn't contains `IndexScan`.
How about `onTable`?